Doc/extending/extending.rst

```diff
@@ -1261,8 +1266,17 @@ function must take care of initializing the C API pointer array::
        /* Create a Capsule containing the API pointer array's address */
        c_api_object = PyCapsule_New((void *)PySpam_API, "spam._C_API", NULL);
 
-       if (c_api_object != NULL)
-           PyModule_AddObject(m, "_C_API", c_api_object);
+       if (c_api_object == NULL) {
```

Comment:
This check is not needed as `PyModule_AddObject()` checks the object for NULL. But you will need to use `Py_XDECREF()` instead of `Py_DECREF()`.